### PR TITLE
Guards - standing around like crash dummies. #2610

### PR DESCRIPTION
**Pull request by bob0bob**
Real problem is the NBT load routine does check to make sure the mob located in the save data still exists. If the mod has been removed since last save or an update to a mob that removed a mob. The mob is deleted MC will crash (exception caught) and many things about MC doesn't work. patrolling, some of the guis.

### CODE REVIEW

**Review by Asherslab — DISMISSED** [review]
Huh, didn't even think of it being possible for the EntityEntry to be null, good spot

**Comment by bob0bob** [comment]
added the braces.

**Review by Asherslab — APPROVED** [review]
Looks good!

**Comment by Asherslab** [comment]
once the checks are all done, i'll merge this if it's good

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 34015 |  0  |  0  |  0   |
| Errors |  593  |  0  |  0  |  0   |
##### More Information:

**Pull request merged**
